src/server/json_family.cc

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/json_family.h"
+
+extern "C" {
+#include "redis/object.h"
+}
+
+#include "base/logging.h"
+#include "server/command_registry.h"
+#include "server/error.h"
+#include "server/tiered_storage.h"
+#include "server/transaction.h"
+
+#include <jsoncons/json.hpp>
+#include <jsoncons_ext/jsonpath/jsonpath.hpp>
+#include <absl/strings/str_join.h>
+
+namespace dfly {
+
+using namespace std;
+using namespace jsoncons;
+
+using CI = CommandId;
+
+namespace {
+
+string GetString(EngineShard* shard, const PrimeValue& pv) {
+  string res;
+  if (pv.IsExternal()) {
+    auto* tiered = shard->tiered_storage();
+    auto [offset, size] = pv.GetExternalPtr();
+    res.resize(size);
+
+    error_code ec = tiered->Read(offset, size, res.data());
+    CHECK(!ec) << "TBD: " << ec;
+  } else {
+    pv.GetString(&res);
+  }
+
+  return res;
+}
+
+OpResult<string> OpGet(const OpArgs& op_args, string_view key, vector<pair<string_view, jsonpath::jsonpath_expression<json>>> expressions) {
```

Comment:
I am sorry, I did not understand wha't the goal of using `vector`. you do not have multiple paths, imho. why what you had before was not good?

Reply:
ignore this comment.